src/index.ts

```diff
@@ -200,6 +209,66 @@ class SvgChunkWebpackPlugin {
 		}
 	}
 
+	/**
+	 * injects the sprite map related to the entry in the related html generated by html-webpack-plugin
+	 * @param {} compilation
+	 * @param {Array<Sprite>} sprites
+	 * @returns {Promise<void>}
+	 */
+	async handleHtmlWebpackPlugin(compilation:Compilation, sprites:Array<Sprite>): Promise<void> {
+		let HtmlWebpackPluginImport;
+		try {
+			// @ts-ignore
+			HtmlWebpackPluginImport = await import('html-webpack-plugin');
+		} catch (e) {
+			HtmlWebpackPluginImport = undefined;
+		}
+		if (HtmlWebpackPluginImport && HtmlWebpackPluginImport.default) {
+			const HtmlWebpackPlugin = HtmlWebpackPluginImport.default;
+			const hooks = HtmlWebpackPlugin.getCompilationHooks(compilation);
+
+			if (this.options.injectSpritesInTemplates) {
```

Comment:
I recommend to move `try/catch` into `if`, because you don't need to make `import`, i.e. `fs`/`network` call if option is disabled